labs/src/main/scala/org/scalalabs/basic/lab02/CollectionExercise.scala

```diff
@@ -50,8 +86,25 @@ object CollectionExercise02 {
    * using a functional approach.
    */
   def groupAdultsPerAgeGroup(persons: Seq[Person]): Map[Int, Seq[Person]] = {
-    error("fix me")
+    var adults: Seq[Person] = (persons.filter((p: Person) => p.age > 17))
+    var sortedAdults: Seq[Person] = adults.sortWith(_.name < _.name)
+    var groupedAdults: List[Seq[Person]] = groupAdults(sortedAdults)
+    var ageGroups: List[Int] = groupAges(groupedAdults)
+    var groupings: List[(Int, Seq[Person])] = ageGroups.zip(groupedAdults)
+    var map = Map(groupings map { pair => pair._1 -> pair._2 }: _*)
+    return map
+  }
+  
+  def groupAdults(persons: Seq[Person]): List[Seq[Person]] = {
+    if (persons.length == 0) return List()
+    else return List(persons.filter((p: Person) => (p.age) / 10 * 10 == ((persons.head).age) / 10 * 10)) ::: groupAdults(persons.filter ((p: Person) => (p.age) / 10 * 10 != ((persons.head).age) / 10 * 10))
+  }
+  
+  def groupAges(personList: List[Seq[Person]]): List[Int] = {
+    if (personList.length == 0) return List()
+    else return List(((personList.head).head).age / 10 * 10) ::: groupAges(personList.tail)
   }
```

Comment:
Nice job!

This did end up being pretty complex though. Perhaps `groupBy` could help?